### PR TITLE
Add full factom did support from identity client

### PR DESCRIPTION
**Pull request by srmalley-sphereon**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -10,7 +10,7 @@
     <parent>
         <groupId>decentralized-identity</groupId>
         <artifactId>uni-resolver</artifactId>
-        <version>0.1.0</version>
+        <version>0.3-SNAPSHOT</version>
```
Comment (nklomp) [inline pom.xml:13]: 0.3.0-SNAPSHOT following SEMVER and Maven versioning

File: src/main/java/uniresolver/driver/did/factom/DIDFactomDriver.java [hunk 2]
```diff
@@ -70,37 +93,40 @@ public ResolveResult resolve(String identifier) throws ResolutionException {
...
-            IdentityResponse identityResponse = getClient(networkId).getIdentityResponse(targetIdentifier, EntryValidation.IGNORE_ERROR, Optional.empty(), Optional.empty());
-            DIDDocument didDocument = getClient(networkId).factory().toDid(identifier, identityResponse);
-            ResolveResult resolveResult = ResolveResult.build(didDocument);
-            resolveResult.setMethodMetadata(createMethodMetadata(identifier, networkId, identityResponse, didDocument, start));
-            resolveResult.setResolverMetadata(createResolverMetadata(identifier, identityResponse, didDocument, start));
-            return resolveResult;
-        } catch (RuleException e) {
+            IdentityClient client = getClient(networkId);
+            List<FactomIdentityEntry<?>> allEntries = client.lowLevelClient()
+                    .getAllEntriesByIdentifier(identifier, EntryValidation.THROW_ERROR, Optional.empty(), Optional.empty());
```
Comment (nklomp) [inline src/main/java/uniresolver/driver/did/factom/DIDFactomDriver.java]: Do we really want to throw errors on an invalid entry? The previous one was ignore errors, which makes more sense to me (skipping invalid entries altogether on resolution, according to the Factom DID spec)

**Review by nklomp — CHANGES_REQUESTED** [review]
Some small remarks

File: pom.xml (hunk 1 above)
Comment (srmalley-sphereon) [reply pom.xml:13]: Yes, however this is not my naming. This is the parent which has a version set by DIF (DanubeTech in this case). https://github.com/decentralized-identity/universal-resolver/blob/main/resolver/java/pom.xml

File: src/main/java/uniresolver/driver/did/factom/DIDFactomDriver.java (hunk 2 above)
Comment (srmalley-sphereon) [reply src/main/java/uniresolver/driver/did/factom/DIDFactomDriver.java]: updated to ignore.

File: pom.xml (hunk 1 above)
Comment (nklomp) [reply pom.xml:13]: Ah LOL. Yupz didn't notice that

Comment (nklomp) [reply pom.xml:13]: But then the 0.1-SNAPSHOT version is ours and has the same problem ;)

**Review by nklomp — CHANGES_REQUESTED** [review]
See version remark

File: pom.xml (hunk 1 above)
Comment (srmalley-sphereon) [reply pom.xml:13]: Oh right, yes I can correct that one.

Comment (srmalley-sphereon) [reply pom.xml:13]: Updated.

**Review by nklomp — APPROVED** [review]

**Pull request merged**